### ISSUE TITLE
Bug: backupStorageLocations name not passed from Velero CR to backupStorageLocation CR

### ISSUE THREAD

**Issue by gorantornqvist** [openshift/oadp-operator#414]
### Describe bug

Hi,
I am trying to setup OADP with multiple teams where each team has different BackupStorageLocations 
But the name set in the Velero CRs spec.backupStorageLocations isnt passed to the BackupStorageLocations CR.
Instead of naming the CR to "team1" it names it to "sf-velero-1" which seem generated by the operator.
```
kind: Velero
metadata:
  name: sf-velero
  namespace: openshift-adp
spec:
  olmManaged: true
  backupStorageLocations:
  # TEAM 1
    - name: team1
      credential:
        key: cloud
        name: cloud-credentials-team1
      provider: aws
      objectStorage:
        bucket: team1
        prefix: someprefix
        caCert: LS0tLS1CRUdJ...
      config:
        region: us-east-1
        profile: "default"
        insecureSkipTLSVerify: "false"
        signatureVersion: "4"
        s3Url: "https://my-objectstorage"
        s3ForcePathStyle: "true"
  defaultVeleroPlugins:
    - aws
    - csi
    - openshift
  veleroFeatureFlags:
    - EnableCSI
  enableRestic: true
```
With over 20 teams I guess it would be hard to follow which BackupStorageLocation belongs to who.

I also tried creating a different Velero CR for the other team but the operator wouldnt allow that and gave this error:
```
Reconciler error        {"reconciler group": "oadp.openshift.io", "reconciler kind": "Velero", "name": "team1", "namespace": "openshift-adp", "error": "Object openshift-adp/restic-restore-action-config is already owned by another Velero controller sf-velero"}
```
So an additional question would be, what is the correct way to do this kind of seperation?

Thanks ...

EDIT:
Looks like the name property has been removed?
https://github.com/openshift/oadp-operator/blob/oadp-0.4.0/deploy/olm-catalog/bundle/manifests/veleroes.crd.yaml#L47

I know "name" worked in the older 0.2.6 version: https://github.com/openshift/oadp-operator/blob/oadp-0.2.6/deploy/crds/konveyor.openshift.io_v1alpha1_velero_cr.yaml#L12

### What happened?

The BackupStorageLocation CR gets a name that I cant control

Running OADP 0.4.0 (but that version isnt available in below dropdown)

### OADP Version

0.3.x (Beta)

### OpenShift Version

4.8

### Velero pod logs

```shell
-
```

### Restic pod logs

```shell
-
```

### Operator pod logs

```shell
-
```

### New issue

- [X] This issue is new

**Comment by kaovilai** [comment]
We currently need a sorted way (currently bslName) to link to unnamed (expected default case) BSLs in OADP CR. We could change bslName dependency to using labels or annotations

**Comment by gorantornqvist** [comment]
I found that migration between clusters according to https://velero.io/docs/v1.7/migration-case/ is really hard when you cant specify the bslName since when adding the original bsl to the migration/destination cluster it will sync backup API-objects that reference a storageLocation with the bslName in the old cluster and these doesnt match the blsName

**Comment by kaovilai** [comment]
@gorantornqvist For migration purposes, we would recommend you look at https://access.redhat.com/documentation/en-us/openshift_container_platform/4.10/html/migration_toolkit_for_containers/index
or https://crane-docs.konveyor.io/

This project on its own do not support migration use case.

**Comment by kaovilai** [comment]
We can look into this in the future but it will not be backported to v0.3

**Comment by kaovilai** [comment]
Resolved via #978
